Modules/FindCUDA.cmake

```diff
@@ -41,11 +41,26 @@
 #      nvcc in the generated source.  If you compile to PTX and then load the
 #      file yourself, you can mix bit sizes between device and host.
 #
+#   CUDA_ARCH_NAME (Default is "Auto" for regular builds and "Manual" for cross-builds)
+#   -- Controls the way CUDA_SELECT_NVCC_ARCH_FLAGS function selects CUDA compute architecture flags.
+#      If you are building for local install, "Auto" should work best as it determines local machine compute arch.
+#      If you are building for distribution, set to "Common" or "All" to cover common subsets of architectures.
+#      "Manual" makes CUDA_SELECT_NVCC_ARCH_FLAGS to get flags directly from ENV{CUDA_ARCH_BIN}
```

Comment:
Do you expect this to read the environment variable every time you configure or just the first time?  It seems messy to rely on the continual presence of this environment variable.

Would it be better to use the environment variable (when defined) to be the default for CUDA_ARCH_NAME?